javascript/react/good-to-know/immutability-helpers-in-react.md

```diff
@@ -25,52 +25,73 @@ aspects:
 
 ---
 
-# Immutability helpers in **React**
+# Immutability helpers in React
 
 ---
 ## Content
 
-Opting out for an immutable style of data management, a significant speed up can be achieved by easily implementing a fast `shouldComponentUpdate()` lifecycle method.
+When opting in for an immutable style of data management, a significant speedup can be achieved by easily implementing a fast `shouldComponentUpdate()` lifecycle method. To do this, you would need a method that aids with comparing *versions* of your component's `state`.
 
-Given the fact that **JavaScript** wasn't designed for dealing with immutable data, **React** provides an immutability helper `update()` with a  syntax inspired by **MongoDB**.
-
-Mutate without having any way of determining which data has changed:
+Let's take a look at the following example of mutating data:
 
 ```javascript
-myData.x.y.z = 7;
-myData.a.b.push(9);
+myData.x.y.z = 10;
+// or...
+myData.a.b.push(13);
+```
+
+Given the fact that **JavaScript** wasn't designed for dealing with immutable data, using one of the above methods does not provide a way of determining which data has changed since the previous copy. To help with this, **React** used to provide an immutability helper (has since become legacy) which is now replaced by the `immutability-helper` library.
+
+This library can be installed using the NPM package manager by using the following command:
+
+```shell
+$ npm install immutability-helper --save
 ```
 
-Using the immutability helper:
+Now, with the help of the immutability helper we would write:
+
 ```javascript
-var update = require('react-addons-update');
-var newData = update(myData, {
-  x: {y: {z: {$set: 7}}},
-  a: {b: {$push: [9]}}
+import update from "immutability-helper";
+
+const newData = update(myData, {
+  x: { y: { z: { $set: 10 } } },
+  // myData.x.y.z = 10;
+  a: { b: { $push: [13 ] } }
+  // myData.a.b.push(13);
 });
 ```
-The `$` is a prefix for keys that are called commands such as `{$push: array}` or `{$apply: function}`.
 
-(Shallow) merge:
+Although this format might need some getting used to, it provides a much better approach for determining which data has changed [1].
+
+In terms of notation, keys that are preceded by a `$` are called **commands** (`{push: array}`, `{apply: function}`, etc.) and the data that is being mutated is called the **target**. 
+
+Here is how you would perform a shallow merge using the `update()` method:
+
 ```javascript
-var obj = {a: 5, b:3};
+var obj = { a: 5, b: 3 };
 var newObj = update(obj, {
-  $merge: {b: 6, c: 7}});
-//newObj = {a: 5, b:6, c:7}
+  $merge: { b: 6, c: 7 }
+});
+//newObj = {a: 5, b: 6, c:7}
 ```
 
-Other worth mentioning intuitive commands are `{$unshift: array}` and `{$splice: array of arrays}`.
+Other worth mentioning intuitive commands are `{$unshift: array}` and `{$splice: array of arrays}`, but you can see all the available methods in the Resources section of this insight.
```

Comment:
Word order seems a bit unnatural here...maybe we should say (and remove "intuitive" cause it's subjective)?

> Other commands worth mentioning are...